src/Resources/contao/modules/ModuleArticle.php

```diff
@@ -366,7 +366,7 @@ public function generatePdf()
 		// Close and output PDF document
 		$pdf->lastPage();
 		$pdf->Output(standardize(ampersand($this->title, false)) . '.pdf', 'D');
-
+		// FIXME: This has to be changed to a ResponseException somehow - rather write to temp file and stream it then?
```

Comment:
IMO we should never echo something as we might have sub requests etc. in the future.